Please check the CI error:

```
Traceback (most recent call last):
  File "/__w/account-financial-reporting/account-financial-reporting/account_financial_report/tests/test_trial_balance.py", line 447, in test_03_partner_balance
    res_data = self._get_report_lines(with_partners=True)
  File "/__w/account-financial-reporting/account-financial-reporting/account_financial_report/tests/test_trial_balance.py", line 203, in _get_report_lines
    res_data = self.env[
  File "/__w/account-financial-reporting/account-financial-reporting/account_financial_report/report/trial_balance.py", line 825, in _get_report_values
    total_amount, accounts_data, partners_data = self._get_data(
  File "/__w/account-financial-reporting/account-financial-reporting/account_financial_report/report/trial_balance.py", line 510, in _get_data
    total_amount, partners_data = self._compute_partner_amount(
  File "/__w/account-financial-reporting/account-financial-reporting/account_financial_report/report/trial_balance.py", line 331, in _compute_partner_amount
    tmp_list = sorted(
  File "/__w/account-financial-reporting/account-financial-reporting/account_financial_report/report/trial_balance.py", line 333, in <lambda>
    key=lambda x: isinstance(x[1], dict) and x[1]["partner_name"] or x[0],
KeyError: 'partner_name'
```

--- a/account_financial_report/report/open_items.py
+++ b/account_financial_report/report/open_items.py
@@ -215,11 +215,18 @@ class OpenItemsReport(models.AbstractModel):
 
     @api.model
     def _order_open_items_by_date(
-        self, open_items_move_lines_data, show_partner_details, partners_data
+        self,
+        open_items_move_lines_data,
+        show_partner_details,
+        partners_data,
+        accounts_data,
     ):
+        # We need to order by account code, partner_name and date
+        accounts_data_sorted = sorted(accounts_data.items(), key=lambda x: x[1]["code"])
+        account_ids_sorted = [account[0] for account in accounts_data_sorted]
         new_open_items = {}
         if not show_partner_details:
-            for acc_id in open_items_move_lines_data.keys():
+            for acc_id in account_ids_sorted:
                 new_open_items[acc_id] = {}
                 move_lines = []
                 for prt_id in open_items_move_lines_data[acc_id]:
@@ -228,7 +235,7 @@ class OpenItemsReport(models.AbstractModel):
                 move_lines = sorted(move_lines, key=lambda k: (k["date"]))
                 new_open_items[acc_id] = move_lines
         else:
-            for acc_id in open_items_move_lines_data.keys():
+            for acc_id in account_ids_sorted:
                 new_open_items[acc_id] = {}
                 for prt_id in sorted(
                     open_items_move_lines_data[acc_id],
@@ -279,7 +286,10 @@ class OpenItemsReport(models.AbstractModel):
 
         total_amount = self._calculate_amounts(open_items_move_lines_data)
         open_items_move_lines_data = self._order_open_items_by_date(
-            open_items_move_lines_data, show_partner_details, partners_data
+            open_items_move_lines_data,
+            show_partner_details,
+            partners_data,
+            accounts_data,
         )
         return {
             "doc_ids": [wizard_id],

--- a/account_financial_report/report/trial_balance.py
+++ b/account_financial_report/report/trial_balance.py
@@ -330,7 +330,10 @@ class TrialBalanceReport(models.AbstractModel):
         for acc_id, total_data in total_amount.items():
             tmp_list = sorted(
                 total_data.items(),
-                key=lambda x: isinstance(x[1], dict) and x[1]["partner_name"] or x[0],
+                key=lambda x: isinstance(x[0], int)
+                and isinstance(x[1], dict)
+                and x[1]["partner_name"]
+                or x[0],
             )
             total_amount[acc_id] = {}
             for key, value in tmp_list:

